datafusion/common/Cargo.toml

```diff
@@ -34,16 +34,16 @@ path = "src/lib.rs"
 
 [features]
 avro = ["apache-avro"]
+default = []
 jit = ["cranelift-module"]
-pyarrow = ["pyo3"]
+pyarrow = ["pyo3", "arrow/pyarrow"]
 
 [dependencies]
-apache-avro = { version = "0.14", features = ["snappy"], optional = true }
-arrow = { version = "23.0.0", features = ["prettyprint"] }
-avro-rs = { version = "0.13", features = ["snappy"], optional = true }
+apache-avro = { version = "0.14", default-features = false, features = ["snappy"], optional = true }
+arrow = { version = "23.0.0", default-features = false }
 cranelift-module = { version = "0.87.0", optional = true }
-object_store = { version = "0.5.0", optional = true }
+object_store = { version = "0.5.0", default-features = false, optional = true }
 ordered-float = "3.0"
-parquet = { version = "23.0.0", features = ["arrow"], optional = true }
+parquet = { version = "23.0.0", default-features = false, optional = true }
```

Comment:
Given the only reason for these dependencies is deriving From conversions for ArrowError, perhaps we could/should just remove them. This is what I have been doing when splitting apart arrow-rs - see https://github.com/apache/arrow-rs/pull/2711.

In general the error handling is a little bit unfortunate, so perhaps this will all get addressed eventually by https://github.com/apache/arrow-rs/issues/2725

Reply:
Yes absolutely. I was planning on revisiting this after the arrow-rs split.